app/src/main/java/com/github/polypoly/app/map/MapActivity.kt

```diff
@@ -136,6 +144,73 @@ class MapActivity : ComponentActivity() {
         }, modifier = Modifier.testTag("map"))
     }
 
+    @Composable
+    fun RollDiceButton() {
+        Box(modifier = Modifier.fillMaxWidth()) {
+            Button(
+                modifier = Modifier
+                    .size(80.dp)
+                    .align(Alignment.BottomCenter)
+                    .offset(y = (-80).dp)
+                    .testTag("rollDiceButton"),
+                onClick = {
+                    // Toast.makeText(this@MapActivity, rollDice(), Toast.LENGTH_SHORT).show()
+                    showRollDiceDialog.value = true
+                },
+                shape = CircleShape
+
+            ) {
+                Icon(Icons.Filled.Casino, contentDescription = "Roll Dice")
+            }
+        }
+    }
+
+    /**
+     * Dice roll dialog, shows the result of 3 dice rolls in a column.
+     */
+    @Composable
+    fun RollDiceDialog() {
+        if (showRollDiceDialog.value) {
+            Dialog(onDismissRequest = { showRollDiceDialog.value = false }) {
+                AlertDialog(
+                    onDismissRequest = { showRollDiceDialog.value = false },
+                    title = { Text("Dice Roll") },
+                    text = {
+                        Column {
+                            // 3 buttons, containing the name of a random location
+                            Button(onClick = { }) {
+                                Text(rollDice().name)
+                            }
+                            Button(onClick = { }) {
+                                Text(rollDice().name)
+                            }
+                            Button(onClick = { }) {
+                                Text(rollDice().name)
+                            }
+                        }
+                    },
+                    confirmButton = {
+                        Button(onClick = { showRollDiceDialog.value = false }) {
+                            Text("Quit")
+                        }
+                    }
+                )
+            }
+        }
+    }
+
+    private fun rollDice(): com.github.polypoly.app.game.Location {
+        val diceRollsSum = IntArray(2) { (1..6).random() }.sum() - 1
```

Comment:
You have to do `- 2` instead of `- 1` because when you add two number between 1 and 6, the range is between 2 and 12, and a tab index begin to 0